datalake/src/main/java/com/sequenceiq/datalake/service/sdx/StackRequestManifester.java

```diff
@@ -269,6 +270,27 @@ void setupCloudStorageAccountMapping(StackV4Request stackRequest, String environ
         }
     }
 
+    void validateMappingsConfig(MappingsConfig mappingsConfig, StackV4Request stackRequest) {
+        String errorMessage = null;
+
+        // Validate RAZ if enabled, making sure that the RAZ mapping exists only when it is required.
+        if (stackRequest.getCluster().isRangerRazEnabled()) {
+            if (!mappingsConfig.getActorMappings().containsKey("rangerraz")) {
+                errorMessage = "IDBMMS mappings must contain the RAZ role if RAZ is to be created!";
+            }
+        } else {
+            if (mappingsConfig.getActorMappings().containsKey("rangerraz")) {
+                errorMessage = "IDBMMS mappings cannot contain the RAZ role if RAZ is not enabled for the cluster!";
+            }
+        }
+
+        // Throw error if validation failed.
+        if (errorMessage != null) {
+            LOGGER.error(errorMessage);
+            throw new IdbmmsOperationException(errorMessage);
```

Comment:
Minor: I would rather directly throw `BadRequestException` since the latter more appropriately expresses that the problem is due to a conflict of cluster settings vs IDBroker mappings. `IdbmmsOperationException` is supposed to  denote problems specific to IDBMMS, e.g. due to missing mappings, communication error or some other internal service error.

Reply:
Badrequest does sound better.